src/AElf.Kernel.Core/Blockchain/Application/IBlockValidationProvider.cs

```diff
@@ -80,28 +81,42 @@ public ValidateNextTimeBlockValidationException(string message, Exception inner)
 
     public class BlockValidationProvider : IBlockValidationProvider
     {
-        public async Task<bool> ValidateBlockBeforeExecuteAsync(IBlock block)
+        private IBlockchainService _blockchainServce;
+        public BlockValidationProvider(IBlockchainService blockchainService)
+        {
+            _blockchainServce = blockchainService;
+        }
+
+        public async Task<bool> ValidateBeforeAttachAsync(IBlock block)
         {
             if (block?.Header == null || block.Body == null)
-            {
                 return false;
-            }
 
             if (block.Body.TransactionsCount == 0)
-            {
                 return false;
-            }
+
+            if (_blockchainServce.GetChainId() != block.Header.ChainId)
+                return false;
+
+            if (block.Height != 1 && !block.VerifySignature())
+                return false;
 
             if (block.Body.CalculateMerkleTreeRoots() != block.Header.MerkleTreeRootOfTransactions)
-            {
                 return false;
-            }
 
-            // TODO: Time span maybe configurable.
-            if (block.Header.Time.ToDateTime() - DateTime.UtcNow > TimeSpan.FromMilliseconds(2000))
-            {
+            if (block.Height != 1 && block.Header.Time.ToDateTime() - DateTime.UtcNow > KernelConsts.AllowedFutureBlockTimeSpan)
```

Comment:
if we have const GenesisBlockHeight, we should use it

Reply:
Fixed